Can't `#[bench]` a function named `b`

```rust
#![feature(test)]

extern crate test;

#[bench]
fn b(_: &mut test::bench::Bencher) {}
```

I should be able to run this as a benchmark, but I got the following error:

```
error[E0308]: mismatched types
 --> src/lib.rs:6:1
  |
5 | #[bench]
  | -------- in this attribute macro expansion
6 | fn b(_: &mut test::bench::Bencher) {}
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | expected `&mut Bencher`, found `TestDescAndFn`
  | expected due to this
  | constant defined here
  | `b` is interpreted as a constant, not a new binding
  | help: introduce a new binding instead: `other_b`

error[E0308]: mismatched types
 --> src/lib.rs:6:1
  |
5 | #[bench]
  | -------- in this attribute macro expansion
6 | fn b(_: &mut test::bench::Bencher) {}
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | expected `&mut Bencher`, found `TestDescAndFn`
  | arguments to this function are incorrect
  |
note: function defined here
 --> src/lib.rs:6:4
  |
6 | fn b(_: &mut test::bench::Bencher) {}
  |    ^ ----------------------------

For more information about this error, try `rustc --explain E0308`.
```

It seems that the macro expansion of `#[bench]` has a name collision with a variable named `b`. https://github.com/rust-lang/rust/blob/292be5c7c05138d753bbd4b30db7a3f1a5c914f7/compiler/rustc_builtin_macros/src/test.rs#L210-L218

### Meta

Reproducible on the playground with version `1.93.0-nightly (2025-10-28 278a90913daf77077910)` (click on the three dots on top left -> "test")

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/compiler/rustc_builtin_macros/src/test.rs
+++ b/compiler/rustc_builtin_macros/src/test.rs
@@ -207,30 +207,30 @@ pub(crate) fn expand_test_or_bench(
     };
 
     let test_fn = if is_bench {
-        // A simple ident for a lambda
-        let b = Ident::from_str_and_span("b", attr_sp);
-
+        // avoid name collisions by using the function name within the identifier, see bug #148275
+        let bencher_param =
+            Ident::from_str_and_span(&format!("__bench_{}", fn_.ident.name), attr_sp);
         cx.expr_call(
             sp,
             cx.expr_path(test_path("StaticBenchFn")),
             thin_vec![
                 // #[coverage(off)]
-                // |b| self::test::assert_test_result(
+                // |__bench_fn_name| self::test::assert_test_result(
                 coverage_off(cx.lambda1(
                     sp,
                     cx.expr_call(
                         sp,
                         cx.expr_path(test_path("assert_test_result")),
                         thin_vec![
-                            // super::$test_fn(b)
+                            // super::$test_fn(__bench_fn_name)
                             cx.expr_call(
                                 ret_ty_sp,
                                 cx.expr_path(cx.path(sp, vec![fn_.ident])),
-                                thin_vec![cx.expr_ident(sp, b)],
+                                thin_vec![cx.expr_ident(sp, bencher_param)],
                             ),
                         ],
                     ),
-                    b,
+                    bencher_param,
                 )), // )
             ],
         )

